lib/daru/index/index.rb

```diff
@@ -44,24 +44,28 @@ def each(&block)
     end
 
     attr_reader :relation_hash, :size
+    attr_accessor :name
 
-    def initialize index
-      index =
-        case index
-        when nil
-          []
-        when Integer
-          index.times.to_a
-        when Enumerable
-          index.to_a
-        else
-          raise ArgumentError,
-            "Cannot create index from #{index.class} #{index.inspect}"
-        end
-
+    # @example
+    #
+    # idx = Daru::Index.new [:one, 'one', 1, 2, :two]
+    # => #<Daru::Index(5): {one, one, 1, 2, two}>
+    #
+    #   # set the name
+    # idx.name = "index_name"
+    # => "index_name"
+    # idx
+    # => #<Daru::Index(5): index_name {one, one, 1, 2, two}>
+    #
+    #   # set the name during initialization
+    # idx = Daru::Index.new [:one, 'one', 1, 2, :two], name: "index_name"
+    # => #<Daru::Index(5): index_name {one, one, 1, 2, two}>
```

Comment:
Have you checked how it is rendered by YARD? You can run `yard server --reload` locally and look at results. (Related to all classes where you've added/updated docs)